### PR TITLE
Persistence state implementation

### PR DESCRIPTION
**Pull request by javanlacerda**
Solve the issue #58 

This PR implements persistence to save jobs states, this following points are implemented:

- Database connectors interface
- Etcd3 connector (plugin)
- Sqlite connector (plugin)
- Synchronize system between job plugin and cluster
- Persist state when changed
- Define the default persistence plugin (sqlite)
- Refactor same docstrings to inside function patter
- Create a util decorator set_timeout
- Add dill, etcd3, peewee and urlib3 to requirements
- Implement tests to new functionalities
- Update documentation

### CODE REVIEW

File: broker/service/api/v10.py [hunk 1]
```diff
@@ -469,9 +469,10 @@ def delete_all_submissions(data):
         ex.UnauthorizedException -- Authetication problem
     """
     check_authorization(data)
+    db_connector.delete_all()
 
-    for id in submissions.keys():
-        delete_submission(id, data)
+    for key in submissions.keys():
```
Comment (clenimar) [inline broker/service/api/v10.py]: no need to call keys() here.

File: broker/utils/timer.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import threading
+from functools import wraps
+
+
+def set_timeout(delay=0.):
+    """
+    Decorator that create a timeout
```
Comment (clenimar) [inline broker/utils/timer.py]: typo: creates. (also you should leave it as a one-line docstring).

**Review by clenimar — COMMENTED** [review]
looking good so far, keep up the good work.

File: broker/service/api/v10.py (hunk 1 above)
Comment (javanlacerda) [reply broker/service/api/v10.py]: In this case it's necessary because when a submission is deleted, the map's length is decremented.

**Comment by matheuscmelo** [comment]
Looks good, just please squash commits and wait for CI

File: broker/plugins/kubejobs/plugin.py [hunk 3]
```diff
@@ -13,53 +13,83 @@
 # limitations under the License.
 
 import datetime
+import json
```
Comment (joseims) [inline broker/plugins/kubejobs/plugin.py]: There should be a line between json and redis import (https://legacy.python.org/dev/peps/pep-0008/#imports)

**Review by matheuscmelo — CHANGES_REQUESTED** [review]
Please squash commits. Besides, LGTM.

**Pull request closed without merging**